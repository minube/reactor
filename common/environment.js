import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
