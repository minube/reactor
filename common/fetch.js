const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
