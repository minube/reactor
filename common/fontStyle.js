import screenSize from './screenSize';
import THEME from './theme';

const { FONT, UNIT } = THEME;

export default (
  width,
  {
    tiny, small, large, title,
  },
) => {
  const {
    TINY, PHONE, TABLET, SMALL, REGULAR, LARGE,
  } = screenSize(width);

  let fontSize = REGULAR || LARGE ? FONT.SIZE.REGULAR : FONT.SIZE.SMALL;

  if (tiny) fontSize = REGULAR || LARGE ? FONT.SIZE.TINY : UNIT * 1.1;
  else if (small) fontSize = REGULAR || LARGE ? FONT.SIZE.SMALL : FONT.SIZE.TINY;
  else if (large) {
    fontSize = FONT.SIZE.LARGE;
    if (TINY) fontSize = FONT.SIZE.SMALL;
    if (PHONE || SMALL) fontSize = FONT.SIZE.REGULAR;
    if (TABLET || REGULAR) fontSize = UNIT * 1.8;
  } else if (title) {
    fontSize = UNIT * 4.8;
    if (TINY || PHONE || TABLET) fontSize = UNIT * 2.4;
    if (SMALL || REGULAR) fontSize = UNIT * 3;
  }

  return {
    fontSize,
    lineHeight: fontSize * (TINY || SMALL ? 1.2 : 1.3),
  };
};
