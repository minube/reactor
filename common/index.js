import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
