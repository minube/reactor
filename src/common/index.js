import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
