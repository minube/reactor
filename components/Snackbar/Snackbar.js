import { func, string } from 'prop-types';
import React from 'react';
import { View } from 'react-native';

import { THEME } from '../../common';
import Button from '../Button';
import Dialog from '../Dialog';
import Text from '../Text';
import styles from './Snackbar.style';

const { COLOR } = THEME;

const Snackbar = ({
  button, caption, onPress, ...inherit
}) => (
  <Dialog visible={inherit.visible} background={false} style={styles.frame} styleContainer={styles.dialog}>
    <View style={styles.container}>
      <Text level={2} style={styles.caption}>
        {caption}
      </Text>
      { button && (
        <Button
          color={COLOR.SECONDARY}
          {...inherit}
          contained={false}
          small
          onPress={onPress}
          style={styles.button}
          title={button.toUpperCase()}
        />)}
    </View>
  </Dialog>
);

Snackbar.propTypes = {
  button: string,
  caption: string,
  onPress: func,
};

Snackbar.defaultProps = {
  button: undefined,
  caption: undefined,
  onPress: undefined,
};

export default Snackbar;
