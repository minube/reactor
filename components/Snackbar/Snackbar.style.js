import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const {
  BORDER_RADIUS, COLOR, UNIT, SPACE,
} = THEME;

export default StyleSheet.create({
  caption: {
    color: COLOR.WHITE,
    flex: 1,
  },

  container: {
    ...LAYOUT.STYLE.ROW,
    marginTop: SPACE.MEDIUM,
  },

  frame: {
    backgroundColor: COLOR.BLACK,
    borderRadius: BORDER_RADIUS,
    minWidth: UNIT * 30,
    maxWidth: UNIT * 51.2,
  },

  dialog: {
    justifyContent: 'flex-end',
    padding: SPACE.MEDIUM,
  },
});
