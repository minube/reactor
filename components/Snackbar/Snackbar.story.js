import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import {
  boolean, color, object, text,
} from '@storybook/addon-knobs/react';

import Snackbar from './Snackbar';

const LOREM = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s.';
const PROPS = { visible: true };

storiesOf('✅ Snackbar', module)
  .add('default', () => (
    <Snackbar {...PROPS} />
  ))
  .add('caption', () => (
    <Snackbar {...PROPS} caption={LOREM} />
  ))
  .add('button', () => (
    <Snackbar {...PROPS} caption={LOREM} button="Close" />
  ))
  .add('visible (false)', () => (
    <Snackbar visible={false} caption={LOREM} />
  ))
  .add('inherit button color', () => (
    <Snackbar {...PROPS} color="red" button="Close" />
  ))
  .add('⚡ onPress', () => (
    <Snackbar {...PROPS} caption={LOREM} button="Close" onPress={action('Snackbar.onPress()')} />
  ))
  .add('🏀 Playground', () => (
    <Snackbar
      caption={text('caption', LOREM)}
      button={text('button', 'Close')}
      onPress={action('Snackbar.onPress()')}
      visible={boolean('visible', true)}
    />
  ));
