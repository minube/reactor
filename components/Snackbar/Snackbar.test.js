import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Snackbar from './Snackbar';

const LOREM = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s.';
const PROPS = { visible: true };
const { COLOR } = THEME;

describe('<Snackbar>', () => {
  it('renders', () => {
    const tree = renderer.create(<Snackbar {...PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<share {...PROPS} caption={LOREM} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {button}', () => {
    const tree = renderer.create(<share {...PROPS} caption={LOREM} button="CLOSE" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<share {...PROPS} caption={LOREM} color={COLOR.ALERT} button="CLOSE" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {visible} (false)', () => {
    const tree = renderer.create(<share visible={false} caption={LOREM} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('when {onPress}', () => {
  //   // @TODO: Test <Snackbar /> {onPress}
  // });
});
