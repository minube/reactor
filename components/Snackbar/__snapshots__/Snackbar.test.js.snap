// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar> renders 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "height": "100%",
      "left": 0,
      "maxHeight": "100%",
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
        Object {
          "justifyContent": "flex-end",
          "padding": 16,
        },
      ]
    }
  >
    <View
      enabled={true}
      keyboardVerticalOffset={0}
      onLayout={[Function]}
      style={
        Array [
          Object {
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minWidth": 320,
          },
          Object {
            "backgroundColor": "#000000",
            "borderRadius": 3,
            "maxWidth": 512,
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "bottom": 0,
            },
          ]
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#000000",
              "borderRadius": 3,
              "maxWidth": 512,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.26,
              "shadowRadius": 5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          />
          <RCTScrollView
            style={
              Array [
                Object {
                  "paddingBottom": 16,
                  "paddingHorizontal": 16,
                },
                false,
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "marginTop": 16,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#4a4a4a",
                        "fontFamily": undefined,
                        "fontWeight": "400",
                        "textAlignVertical": "center",
                      },
                      Object {
                        "fontSize": 14,
                        "lineHeight": 24,
                      },
                      false,
                      Object {
                        "color": "#ffffff",
                        "flex": 1,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`<Snackbar> when {button} 1`] = `
<share
  button="Close"
  caption="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s."
  visible={true}
/>
`;

exports[`<Snackbar> when {caption} 1`] = `
<share
  caption="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s."
  visible={true}
/>
`;

exports[`<Snackbar> when {visible} (false) 1`] = `
<share
  caption="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s."
  visible={false}
/>
`;

exports[`<Snackbar> when inherit {button} color 1`] = `
<share
  button="Close"
  caption="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s."
  color="red"
  visible={true}
/>
`;
